app/javascript/joinGame.js

```diff
@@ -0,0 +1,28 @@
+/* global angular */
+import {web3, Chess} from '../../contract/Chess.sol';
+angular.module('dappChess').controller('JoinGameCtrl',
+  function ($rootScope, $scope) {
+    $scope.availableAccounts = web3.eth.accounts;
+    $scope.selectedAccount = web3.eth.defaultAccount;
+    $scope.username = null;
+    $scope.gameId = null;
+
+    $scope.isSelectedAccount = function (account) {
+      return $scope.selectedAccount === account;
+    };
+    $scope.selectAccount = function (account) {
+      $scope.selectedAccount = account;
+    };
+
+    function joinGame() {
+      $rootScope.$broadcast('message', 'Trying to join the game, please wait a moment...',
+                            'loading', 'joingame');
+      Chess.joinGame($scope.gameId, $scope.username, {from: $scope.selectedAccount});
```

Comment:
- This does not seem to catch the exception that is thrown when it is impossible to join this game (because it already has 2 players).